federationtypes.go

```diff
@@ -448,6 +448,12 @@ func (r RespSendJoin) MarshalJSON() ([]byte, error) {
 	return json.Marshal(partialJoinFields)
 }
 
+// A RespSendKnock is the content of a response to PUT /_matrix/federation/v2/send_knock/{roomID}/{eventID}
+type RespSendKnock struct {
+	// A list of stripped state events to help the initiator of the knock identify the room.
+	KnockRoomState EventJSONs `json:"knock_room_state"`
```

Comment:
Thinking some more about this, the fields from the spec are `content`, `state_key`, `sender` and `type` which is the exact same shape as the `InviteV2StrippedState` type. If you want to be able to access those fields without having to run through the JSON unmarshaller in your tests, then it might be a better idea to define this as:
```
KnockRoomState []InviteV2StrippedState `json:"knock_room_state"`
```